stories/core/Table.stories.tsx

```diff
@@ -718,3 +722,158 @@ export const NoData: Story<TableProps> = ({ columns, data, ...rest }) => {
 NoData.args = {
   data: [],
 };
+
+export const HighlightedSearch: Story<TableProps> = ({
+  columns,
+  data,
+  ...rest
+}) => {
+  type TableStoryDataType = {
+    index: number;
+    name: string;
+    description: string;
+    ids: number[];
+    date: Date;
+  };
+
+  const formatter = useMemo(
+    () =>
+      new Intl.DateTimeFormat('en-us', {
+        month: 'short',
+        day: 'numeric',
+        year: 'numeric',
+      }),
+    [],
+  );
+
+  const formatDate = useCallback(
+    (date: Date) => {
+      return formatter.format(date);
+    },
+    [formatter],
+  );
+
+  const [input, setInput] = React.useState('');
+
+  const regex = useMemo(() => {
+    if (input === '') {
+      return undefined;
+    }
+    /** translate an asterisk-wildcard search to a regex
+     * e.g. "Mic*l" matches "Michael" but not "Michae l"
+     */
+    const compileSimpleSearchQuery = (query: string) => {
+      const escapeChr = (chr: string, str: string) =>
+        str.replace(new RegExp(`\\${chr}`, 'g'), `\\${chr}`);
+      const regexMetacharList = Array.from('\\$^()[]{}|?.+:');
+      const escapedQuery = regexMetacharList.reduce(
+        (prev, chr) => escapeChr(chr, prev),
+        query,
+      );
+      // convert wildcard character to valid regex repeated non-space text definition
+      const queryWithWildcards = escapedQuery.replace(/\*/g, '\\S*');
+      return RegExp(queryWithWildcards, 'gi');
+    };
+    return compileSimpleSearchQuery(input);
+  }, [input]);
+
+  const defaultColumn: Partial<Column> = React.useMemo(
+    () => ({
+      Cell: HighlightingCell,
```

Comment:
If the user has to use a custom cell, then I wonder what is the point of adding a new prop to `Table`.

It should be fairly trivial for the user to handle the highlighting on their end (since `HighlightingCell` just adds `<mark>` with a color), without needing any changes to the Table.

Reply:
perhaps the "feature" I'm adding is just prop clutter and maybe downgrading this PR to just a story showing how to make such a column is better?

Reply:
Yeah I think that would make more sense. But can you confirm that it works without needing any changes to Table? Re-rendering memoized rows is the only thing I can think of that might not work.

Reply:
Maybe still providing/exporting the `HighlightingCell` is good. I'm new to react-table@7 but I'll look into how else to pass the regex state to the cell without a prop on the table

Reply:
yeah this would only work if I made the cell render a stateful component that re-rendered itself through some hacks. Perhaps you there should be some `cellDependencies` prop?

Reply:
I am still investigating what can we do about re-renders. From the first sight it seems that it isn't working for you without any changes because you are using `defaultColumn` prop and we are not checking it for row re-renders. I will write a comment later what I was able to find.

Reply:
defaultColumn probably should not be changing and should never cause re-renders, it is memoized here outside of the `Table` component to prevent that. But the Cell render method does need access to the changing `highlightRegex` probably as a prop. I think it would be difficult to memoize anything if we let defaultColumn change every time `highlightRegex` changes.

Reply:
Hey @MichaelBelousov, sorry for such a late response. We currently don't know what is the best way to handle table changes that are causing re-rendering. Here is a codesandbox with a working cell highlight and it doesn't require any changes to the Table: https://codesandbox.io/s/itwinui-react-minimal-example-forked-5qv8k

Reply:
the `tableColumns` not being memoized make me worried a bit about how it would perform compared to the prop-aware table implementation. When I have some time I think I will just turn this PR into removing any changes I made to the table and making a story using the changes you made in that sandbox. If people have actual performance issues using it then we can look into it again